### PR TITLE
feat: add support for PWA

### PR DESCRIPTION
**Pull request by wei-lee**
https://issues.jboss.org/browse/INTLY-3566

This PR fixes 2 things:

1. Add support for PWA. Turns out we don't have it yet and the app won't work if the device is set to offline.
2. Remove the warning about circular dependencies between the auth service module and the sync service module.

To verify:

1. Run the app using `ionic serve --prod`
2. Login with `test/123` and try view and create some tasks (You can do this on a mobile device as well).
3. Use the network tab to turn offline (or put the app into airplane mode).
4. You should still be able to use the app and continue creating and viewing tasks.
5. Bring the device online again and you should see the data is get synced.

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -54,17 +55,19 @@
               "sourceMap": false,
               "extractCss": true,
               "namedChunks": false,
-              "aot": true,
+              "aot": false,
```
Comment (wei-lee) [inline angular.json]: we need to turn off `aot` compilation for this app in production mode. For some reason when using `aot`, the manage tasks view won't work and there is an error print out in the console. Something to do with this: https://github.com/angular/angular/issues/23878

**Comment by wtrocki** [comment]
An amazing change! Reviewing now

**Comment by wtrocki** [comment]
- `ionic build --prod` is required to generate service worker
- `ionic serve` is not affected and works as it was before 
-  auth works fine

**Comment by wei-lee** [comment]
@wtrocki thanks, I will made the changes you recommended.

**Comment by wei-lee** [comment]
@wtrocki I addressed your comments. I added a new command to make it simple to run the app as PWA app. Also I used `serve` instead of `ionic serve`. I am not sure how to get `ionic serve` to just serving the content from the `www` directory.

File: src/mobile-services.json [hunk 2]
```diff
@@ -1,6 +1,41 @@
 {
-  "version": 1,
-  "namespace": "voyager",
-  "clientId": "voyager-ionic-example",
-  "services": []
+  "clientId": "showcase-app-2",
```
Comment (darahayes) [inline src/mobile-services.json]: Maybe just revert the changes in this file and we're all good!
